soletta/builder.go

```diff
@@ -0,0 +1,42 @@
+package soletta
+
+/*
+#include <soletta.h>
+#include <sol-flow-builder.h>
+*/
+import "C"
+import "unsafe"
+
+type FlowBuilder struct {
+	builder *C.struct_sol_flow_builder
+}
+
+func (fb *FlowBuilder) Init() {
+	fb.builder = C.sol_flow_builder_new()
+}
+
+func (fb *FlowBuilder) AddNode(name string, nodeType string) {
+	cname, cnodeType := C.CString(name), C.CString(nodeType)
+	defer C.free(unsafe.Pointer(cname))
+	defer C.free(unsafe.Pointer(cnodeType))
+	C.sol_flow_builder_add_node_by_type(fb.builder, cname, cnodeType, nil)
+}
+
+func (fb *FlowBuilder) Connect(name1, port1, name2, port2 string) {
+	cname1, cport1, cname2, cport2 := C.CString(name1), C.CString(port1), C.CString(name2), C.CString(port2)
+	defer C.free(unsafe.Pointer(cname1))
+	defer C.free(unsafe.Pointer(cport1))
+	defer C.free(unsafe.Pointer(cname2))
+	defer C.free(unsafe.Pointer(cport2))
+	C.sol_flow_builder_connect(fb.builder, cname1, cport1, -1, cname2, cport2, -1)
+}
+
+func (fb *FlowBuilder) GetNodeType() FlowNodeType {
+	ret := FlowNodeType{}
+	ret.nodeType = C.sol_flow_builder_get_node_type(fb.builder)
+	return ret
+}
+
+func (fb *FlowBuilder) Destroy() {
+	C.sol_flow_builder_del(fb.builder)
+}
```

Comment:
Please take some time to check if you can use finalizer https://golang.org/pkg/runtime/#SetFinalizer and get off this method.

It seems that is ok using it in this case and this would make the things easier for who is using.